src/ripple/ledger/impl/RawStateTable.cpp

```diff
@@ -0,0 +1,214 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/ledger/detail/RawStateTable.h>
+#include <ripple/basics/contract.h>
+
+namespace ripple {
+namespace detail {
+
+// Base invariants are checked by the base during apply()
+
+void
+RawStateTable::apply (RawView& to) const
+{
+    to.rawDestroyXRP(dropsDestroyed_);
+    for (auto const& elem : items_)
+    {
+        auto const& item = elem.second;
+        switch(item.first)
```

Comment:
I'm not sure I follow, you're suggesting an abstract base class? I think the `switch` statement is simpler and easier to audit when reviewing each function to make sure it does the right thing (you did that, I hope). I intentionally wrote it with `switch` statements, and make sure to mention each possible `case` label, to provide more clarity than what would be visible using `if`/`else`